src/serializer/Referentializer.js

```diff
@@ -89,14 +89,35 @@ export class Referentializer {
     // One switch case for one scope.
     const cases = [];
     const serializedScopes = this._getReferentializationState(referentializationScope).serializedScopes;
+    type InitializationCase = {|
+      scopeIDs: Array<number>,
+      values: Array<BabelNodeExpression>,
+    |};
+    const initializationCases: Array<InitializationCase> = [];
     for (const scopeBinding of serializedScopes.values()) {
-      const scopeObjectExpression = t.arrayExpression((scopeBinding.initializationValues: any));
-      cases.push(
-        t.switchCase(t.numericLiteral(scopeBinding.id), [
-          t.expressionStatement(t.assignmentExpression("=", captured, scopeObjectExpression)),
-          t.breakStatement(),
-        ])
-      );
+      const i = initializationCases.findIndex(ic => {
+        return AreSameInitializationValues(ic.values, scopeBinding.initializationValues);
```

Comment:
Time complexity is pretty bad if all cases are different from each other. The other alternative I thought about is to use Map with the `initializationValues` as a key, but it needs to be converted to a string. it's possible because all arguments are numbers.

Reply:
Indeed, this avoidable quadratic behavior is potentially problematic.
At the very minimum, let's leave behind a TODO. 
In practice, I think it is okay for now, and there are other perf bottleness (cough babel traverse cough).

Reply:
Converting the expressions to strings for keying should be easy via `babel-generator`. I think you can do that.

Reply:
Got it. Thanks!